src/bitExpert/Pathfinder/Middleware/RoutingMiddleware.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * This file is part of the Pathfinder package.
+ *
+ * (c) bitExpert AG
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace bitExpert\Pathfinder\Middleware;
+
+use bitExpert\Pathfinder\Router;
+use bitExpert\Pathfinder\RoutingResult;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+class RoutingMiddleware
+{
+    /**
+     * @var Router
+     */
+    protected $router;
+    /**
+     * @var String
+     */
+    protected $attributeName;
+
+    /**
+     * @param Router $router
+     */
+    public function __construct(Router $router)
+    {
+        $this->router = $router;
+        $this->attributeName = RoutingResult::class;
+    }
+
+    /**
+     * Sets the name of the request attribute the routing result will be stored in
+     *
+     * @param $attributeName
+     */
+    public function setAttributeName($attributeName)
```

Comment:
Why not passing the attribute name via the constructor when creating the RoutingMiddleware instance? Maybe we should provide a getter method to be able to programmatically retrieve the current name. Otherwise you would need to hardcode the name somewhere in your codebase, maybe even in 2 different locations.